keycloak/providers/authenticator/hmda/src/main/java/gov/cfpb/keycloak/authenticator/hmda/HmdaValidInstitutionsFormAction.java

```diff
@@ -0,0 +1,268 @@
+package gov.cfpb.keycloak.authenticator.hmda;
+
+import org.jboss.logging.Logger;
+import org.keycloak.Config;
+import org.keycloak.authentication.FormAction;
+import org.keycloak.authentication.FormActionFactory;
+import org.keycloak.authentication.FormContext;
+import org.keycloak.authentication.ValidationContext;
+import org.keycloak.authentication.forms.RegistrationPage;
+import org.keycloak.events.Errors;
+import org.keycloak.forms.login.LoginFormsProvider;
+import org.keycloak.models.*;
+import org.keycloak.models.utils.FormMessage;
+import org.keycloak.provider.ConfiguredProvider;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.services.validation.Validation;
+
+import javax.net.ssl.*;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import java.net.URI;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+
+public class HmdaValidInstitutionsFormAction implements FormAction, FormActionFactory, ConfiguredProvider {
+
+    private static final WebTarget apiClient;
+    public static final String PROVIDER_ID = "registration-institution-action";
+    public static final String FIELD_INSTITUTIONS = "user.attributes.institutions";
+    private static final Logger logger = Logger.getLogger(HmdaValidInstitutionsFormAction.class);
+
+    static {
+        //FIXME: Should probably switch envvar handling to using Keycloak built-in Config features
+        String validateSsl = System.getenv("INSTITUTION_SEARCH_VALIDATE_SSL");
+        String apiUri = System.getenv("INSTITUTION_SEARCH_URI");
+
+        ClientBuilder apiClientBuilder = ClientBuilder.newBuilder();
+
+        // Special handling for dealing with untrusted HTTPS calls
```

Comment:
This should only come into play for local development.  Initially I set this up with no HTTPS for dev, but since I had to do an AJAX call from Keycloak's registration page (which is HTTPS), I had to make this call HTTPS too.

Once we move out of local dev, we will have real certs in place, so this should no longer come into play.  In those envs, we'll switch to [`INSTITUTION_SEARCH_VALIDATE_SSL: "ON"`](https://github.com/cfpb/hmda-platform/pull/687/files#diff-4e5e90c6228fd48698d074241c2ba760R37), which should skip this setup, and enforce SSL cert validation.